src/vmm/src/device_manager/mmio.rs

```diff
@@ -142,15 +143,24 @@ impl MMIODeviceManager {
         resource_allocator: &mut ResourceAllocator,
         irq_count: u32,
     ) -> Result<MMIODeviceInfo, MmioError> {
-        let irqs = resource_allocator.allocate_gsi(irq_count)?;
+        if irq_count != 0 && irq_count != 1 {
+            return Err(MmioError::InvalidIrqConfig);
+        }
+        let irqs_vec = resource_allocator.allocate_gsi(irq_count)?; 
+        let irq : Option<NonZeroU32> = if irqs_vec.len() == 0 {
+            None
+        } else {
+            NonZeroU32::new(irqs_vec.get(0).unwrap().clone())
+        };
```

Comment:
Slightly simpler (albeit less efficient on the error path, but, its the error path, so we don't really care)

```suggestion
        let irq = match &resource_allocator.allocate_gsi(irq_count)? {
            &[] => None, 
            &[irq] => NonZeroU32::new(irq),
            _ =>  return Err(MmioError::InvalidIrqConfig)
        };
```